index.js

```diff
@@ -386,6 +391,81 @@ exports.getLastThreadPost = function (thread) {
   return msg
 }
 
+/* post revision utils */
+
+exports.getRevisions = function(ssb, thread, callback) {
+  function collectRevisions(thread) {
+    // this function walks the revisions of a given thread, collecting them up
+    // asyncly
+
+    var threadRevisions = thread.related
+        .map(function(relatedMsg) {
+      
+        if (relatedMsg.value.content.type === 'post-edit' &&
+            // ^ make sure it's an edit
+            relatedMsg.value.content.revision &&
+            // ^ may be unnecessary if schema validates
+            (relatedMsg.value.content.revision === thread.key ||
+             relatedMsg.value.content.root     === thread.key)
+            // ^ either the msg revises the root, or it's a subsequent revision
+           ) {
+          return(relatedMsg);
+        }
+      })
+      .filter(function(msg) { return msg || false}) // eliminate falsehood
+      .sort(function (oneRev, otherRev) { // remove duplicates by sorting and
+                                          // reducing over the sorted arr
+        return oneRev.key < otherRev.key
+      });
+    if (threadRevisions.length > 0) {
+      threadRevisions =
+        threadRevisions.reduce(function(prevRevs, thisRev, thisInd) {
+          var previousKey = "";
+          prevRevs.length ?
+            previousKey = prevRevs[prevRevs.length - 1].key :
+            previousKey = prevRevs.key
+          if (previousKey === thisRev.key) {
+            // remove duplicate
+            return (prevRevs instanceof Array ? prevRevs : [prevRevs])
+          } else {
+            if (!prevRevs.concat) { // js, your reduce is strange
+              return [].concat(prevRevs, thisRev)
+            } else {
+              return prevRevs.concat(thisRev)
+            }
+          }
+        })
+    }
+    return threadRevisions
+  }
+            
+
+  if (!thread.hasOwnProperty('related')) { // if the thread doesn't have its related objects,
+                         // fetch them
+    ssb.relatedMessages(thread, function(err, enrichedThread) {
+      if (err) throw err
+      else if (!enrichedThread.hasOwnProperty('related')) callback([]) // if still no related objects
+      else callback(collectRevisions(enrichedThread))
+    })
+  } else { // note: this branch is technically synchronous and the above is not
+           // :(
+    callback(collectRevisions(thread))
```

Comment:
Use the node `(err, result)` callback convention. Except in rare cases, we try to use that in all situations.